pkg/controller/daemon/update.go

```diff
@@ -171,6 +194,14 @@ func (dsc *DaemonSetsController) rollingUpdate(ctx context.Context, ds *apps.Dae
 				// no point considering any other candidates
```

Comment:
hm, just noticed one thing: we should move this `numSurge >= maxSurge` check after the `!shouldRun` check bellow, since sunsetting nodes is not dependent on it and we can delete pods sooner

Reply:
I think it makes sense, so I'll fix it.